components/service/glean/src/main/java/mozilla/components/service/glean/storages/StringListsStorageEngine.kt

```diff
@@ -0,0 +1,133 @@
+import mozilla.components.service.glean.CommonMetricData
+import mozilla.components.support.base.log.logger.Logger
+import org.json.JSONArray
+import java.lang.Exception
+
+/**
+ * This singleton handles the in-memory storage logic for string lists. It is meant to be used by
+ * the Specific String List API and the ping assembling objects. No validation on the stored data
+ * is performed at this point: validation must be performed by the Specific String List API.
+ *
+ * This class contains a reference to the Android application Context. While the IDE warns
+ * us that this could leak, the application context lives as long as the application and this
+ * object. For this reason, we should be safe to suppress the IDE warning.
+ */
+@SuppressLint("StaticFieldLeak")
+internal object StringListsStorageEngine : StringListsStorageEngineImplementation()
+
+internal open class StringListsStorageEngineImplementation(
+    override val logger: Logger = Logger("glean/StringsListsStorageEngine")
+) : GenericScalarStorageEngine<List<String>>() {
+
+    companion object {
+        // Maximum length of any list
+        const val MAX_LIST_LENGTH_VALUE = 20
+    }
+
+    override fun deserializeSingleMetric(value: Any?): List<String>? {
+        // Since SharedPreferences doesn't directly support storing of List<> types, we must use
+        // an intermediate JSONArray which can be deserialized and converted back to List<String>.
+        // Using JSONArray introduces a possible issue in that it's constructor will still properly
+        // convert a stringified JSONArray into an array of Strings regardless of whether the values
+        // have been properly quoted or not.  For example, [1,2,3] is as valid just like
+        // ["a","b","c"] is valid.
+        return (value as? String)?.let {
+            val jsonArray: JSONArray
+            try {
+                jsonArray = JSONArray(it)
+            } catch (e: org.json.JSONException) {
+                return@let null
+            }
+
+            // The JSONArray creation was successful, now copy items into a new List<> for return.
+            val mutList = mutableListOf<String>()
+            for (i in 0 until jsonArray.length()) {
+                mutList.add(jsonArray.get(i).toString())
+            }
+
+            return@let mutList.asListOfType<String>()
+        }
+    }
+
+    @Suppress("TooGenericExceptionCaught")
+    override fun serializeSingleMetric(
+        userPreferences: SharedPreferences.Editor?,
+        storeName: String,
+        value: List<String>
+    ) {
+        // Since SharedPreferences doesn't directly support storing of List<> types, we must use
+        // an intermediate JSONArray which can be serialized to a String type and stored.
+        try {
+            val jsonArray = JSONArray()
+            value.forEach {
+                jsonArray.put(it)
+            }
+            userPreferences?.putString(storeName, jsonArray.toString())
```

Comment:
Looks like `putString` [doesn't throw](https://developer.android.com/reference/android/content/SharedPreferences.Editor.html#putString(java.lang.String,%20java.lang.String)). Let's get rid of the try/catch